python/lsst/ip/diffim/diffimTools.py

```diff
@@ -307,7 +307,10 @@ def sourceToFootprintList(candidateInList, templateExposure, scienceExposure, ke
 
     for kernelCandidate in candidateInList:
         if not type(kernelCandidate) == afwTable.SourceRecord:
-            raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
+            try:
+                kernelCandidate = kernelCandidate.getSource()
+            except:
+                raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
```

Comment:
The error message is incomplete, since the problem is that the kernelCandidate is neither a SourceRecord nor an object (what kind?) that has a getSource() method.

Cadiate -> Candidate

Do you expect all candidates to be of the same type? If so, I think it would be safer and cleaner to preprocess candidateInList based on the type of the first entry (e.g. convert it to a list of source records, regardless of what it started out as).